cmake/extensions.cmake

```diff
@@ -0,0 +1,370 @@
+include(CheckCCompilerFlag)
+
+########################################################
+# CMake-generic extensions
+########################################################
+
+# Conditionally executing CMake functions with oneliners
+# e.g. if(x) sqrt(10); would be sqrt_ifdef(x 10)
+#
+# <function-name>_ifdef()
+function(add_subdirectory_ifdef feature_toggle dir)
+  if(${${feature_toggle}})
+    add_subdirectory(${dir})
+  endif()
+endfunction()
+
+function(target_sources_ifdef feature_toggle target scope item)
```

Comment:
Can we add comments that explain each one of these functions.

Reply:
Agreed. Will push fix soon.

Reply:
@galak : Rewritten: https://github.com/SebastianBoe/zephyr/blob/e0667a7bcae40722f96a01b8ea691db8e3b4fae9/cmake/extensions.cmake